pyleros/decoder.py

```diff
@@ -29,126 +29,132 @@ def pyleros_decoder(instr_hi, out_sig):
         indls: Signal if the instructio is a indirect load/store
 
     Parameters:
-        None
-
+        debug: Debugging mode, the processor prints various error messages
+        
     """
     
     @always_comb
     def decoder():
 
+        if debug:
+            print("Inside decoder with instruction_hi:", instr_hi)
+
         # Set the defaults of all signals
-        out_sig[int(t_decSignal.op)].next = alu_op_type.LD
-        out_sig[int(t_decSignal.al_ena)].next = False
-        out_sig[int(t_decSignal.ah_ena)].next = False
-        out_sig[int(t_decSignal.log_add)].next = False
-        out_sig[int(t_decSignal.add_sub)].next = False
-        out_sig[int(t_decSignal.shr)].next = False
-        out_sig[int(t_decSignal.sel_imm)].next = False
-        out_sig[int(t_decSignal.store)].next = False
-        out_sig[int(t_decSignal.outp)].next = False
-        out_sig[int(t_decSignal.inp)].next = False
-        out_sig[int(t_decSignal.br_op)].next = False
-        out_sig[int(t_decSignal.jal)].next = False
-        out_sig[int(t_decSignal.loadh)].next = False
-        out_sig[int(t_decSignal.indls)].next = False
+        alu_op.next = alu_op_type.NOP
+        out_sig[int(dec_op_type.al_ena)].next = False
+        out_sig[int(dec_op_type.ah_ena)].next = False
+        out_sig[int(dec_op_type.log_add)].next = False
+        out_sig[int(dec_op_type.add_sub)].next = False
+        out_sig[int(dec_op_type.shr)].next = False
+        out_sig[int(dec_op_type.sel_imm)].next = False
+        out_sig[int(dec_op_type.store)].next = False
+        out_sig[int(dec_op_type.outp)].next = False
+        out_sig[int(dec_op_type.inp)].next = False
+        out_sig[int(dec_op_type.br_op)].next = False
+        out_sig[int(dec_op_type.jal)].next = False
+        out_sig[int(dec_op_type.loadh)].next = False
+        out_sig[int(dec_op_type.indls)].next = False
 
 
         # Decode
+        ins_ckh = instr_hi & 0xf8
 
-        # the imm. bit, 0
-        out_sig[int(t_decSignal.sel_imm)].next = instr_hi[0]
+        # if the instructions are not branch
+        if not (ins_ckh == 0x48):
+            # the imm. bit, 0
+            out_sig[int(dec_op_type.sel_imm)].next = instr_hi[0]
 
-        ins_ckh = instr_hi & 0xf8
+        
 
         if ins_ckh == 0x00:
             # NOP
             pass
 
         elif ins_ckh == 0x08:
             # ADD/SUB
-            out_sig[int(t_decSignal.al_ena)].next = True
-            out_sig[int(t_decSignal.ah_ena)].next = True
-            out_sig[int(t_decSignal.log_add)].next = True
-            out_sig[int(t_decSignal.add_sub)].next = instr_hi[2]
+            out_sig[int(dec_op_type.al_ena)].next = True
+            out_sig[int(dec_op_type.ah_ena)].next = True
+            out_sig[int(dec_op_type.log_add)].next = True
+            out_sig[int(dec_op_type.add_sub)].next = instr_hi[2]
 
```

Comment:
I'm not sure, and perhaps someone will contradict me on this, but you might find this won't convert. I think it is an omission of MyHDL at the moment that there is no integer mappable enum, and is something I've tried in the past to evoke some interest in. Anyway, this might not be possible, in which case the solution might well have to be using constants. :( Anyway, you'll find out!